app/lib/shared/email.dart

```diff
@@ -0,0 +1,137 @@
+
+import 'package:meta/meta.dart';
+
+final _nameEmailRegExp = new RegExp(r'^(.*)<(.+@.+)>$');
+final _defaultFrom = new EmailAddress('Pub Site Admin', 'pub@dartlang.org');
+
+/// Represents a parsed e-mail address.
+class EmailAddress {
+  final String name;
+  final String email;
+
+  EmailAddress(this.name, this.email);
+
+  factory EmailAddress.parse(String value) {
+    value = value.trim();
+    String name = value;
+    String email;
+
+    final match = _nameEmailRegExp.matchAsPrefix(value);
+    if (match != null) {
+      name = match.group(1).trim();
+      email = match.group(2).trim();
+    } else if (value.contains('@')) {
+      final List<String> parts = value.split(' ');
+      for (int i = 0; i < parts.length; i++) {
+        if (parts[i].contains('@') &&
+            parts[i].contains('.') &&
+            parts[i].length > 4) {
+          email = parts[i];
+          parts.removeAt(i);
+          name = parts.join(' ');
+          break;
+        }
+      }
+    }
+    if (name != null && name.isEmpty) {
+      name = null;
+    }
+    if (email != null && email.isEmpty) {
+      email = null;
+    }
+    return new EmailAddress(name, email);
+  }
+
+  bool get isEmpty => name == null && email == null;
+
+  @override
+  String toString() {
+    if (isEmpty) return null;
+    if (email == null) return name;
+    if (name == null) return email;
+    return '$name <$email>';
+  }
+}
+
+/// Represents an email message the site will send.
+class EmailMessage {
+  final EmailAddress from;
+  final List<EmailAddress> recipients;
+  final String subject;
+  final String bodyText;
+
+  EmailMessage(this.from, this.recipients, this.subject, String bodyText)
+      : bodyText = reflowBodyText(bodyText);
+}
+
+/// Parses the body text and splits the [input] paragraphs to [lineLength]
+/// character long lines.
+String reflowBodyText(String input, {int lineLength = 72}) {
+  Iterable<String> reflowLine(String line) sync* {
+    if (line.isEmpty) {
+      yield line;
+      return;
+    }
+    while (line.length > lineLength) {
+      int firstSpace = line.lastIndexOf(' ', lineLength);
+      if (firstSpace < 20) {
+        firstSpace = line.indexOf(' ', lineLength);
+      }
+      if (firstSpace != -1) {
+        yield line.substring(0, firstSpace);
+        line = line.substring(firstSpace).trim();
+      } else {
+        yield line;
+        return;
+      }
+    }
+    if (line.isNotEmpty) {
+      yield line;
+    }
+  }
+
+  return input.split('\n').expand(reflowLine).join('\n');
+}
+
+/// Creates the [EmailMessage]s that we be sent on new package upload.
+EmailMessage createPackageUploadedEmail({
+  @required String packageName,
+  @required String packageVersion,
+  @required String uploaderEmail,
+  @required List<EmailAddress> authorizedUploaders,
+  @required List<EmailAddress> authors,
+}) {
+  final subject = 'Package upload on pub: $packageName $packageVersion';
+  final bodyText = new StringBuffer()
```

Comment:
Couldn't we just use a multi-line string in dart?

Reply:
Yeah, it started out as a bit more complex template, now a multi-line string could be just fine. Changing it...